file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -130,30 +129,108 @@ public static RelDataType deduceRowType(JavaTypeFactory typeFactory,
+                  true);
+              break;
+            case "byte":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.BYTE.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "char":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.CHAR.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "short":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.SHORT.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "int":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.INT.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "long":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.LONG.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "float":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.FLOAT.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "double":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(Primitive.DOUBLE.getBoxClass()).getSqlTypeName()),
+                  true);
+              break;
+            case "date":
+              fieldType = typeFactory.createTypeWithNullability(typeFactory
+                      .createSqlType(typeFactory.createJavaType(java.sql.Date.class)
+                          .getSqlTypeName()),
+                  true);
+              break;
+            case "time":
+              fieldType = typeFactory.createTypeWithNullability(typeFactory
+                      .createSqlType(typeFactory.createJavaType(java.sql.Time.class)
+                          .getSqlTypeName()),
+                  true);
+              break;
+            case "timestamp":
+              fieldType = typeFactory.createTypeWithNullability(
+                  typeFactory.createSqlType(
+                      typeFactory.createJavaType(java.sql.Timestamp.class).getSqlTypeName()),
+                  true);
+              break;
+            default:
+              System.out.println("WARNING: Found unknown type: "
```

Comment:
I think this should be converted into a logging statement as opposed to a system.out (I know it was here before this change but we should clean this kind of thing up when working on the code.)

Reply:
Done.